src/rust.rs

```diff
@@ -463,6 +464,185 @@ macro_rules! rooted {
     }
 }
 
+use jsapi::AutoGCRooter_jspubtd_h_unnamed_1 as AutoGCRooterTag;
+impl AutoGCRooter {
+    pub fn new_unrooted(tag: AutoGCRooterTag) -> AutoGCRooter {
+        AutoGCRooter {
+            down: ptr::null_mut(),
+            tag_: tag as isize,
+            stackTop: ptr::null_mut(),
+        }
+    }
+
+    pub unsafe fn add_to_root_stack(&mut self, cx: *mut JSContext) {
+        let autoGCRooters: &mut _ = {
```

Comment:
Yes, because we need a pointer to the `autoGCRooters_` stack list itself (which is also represented as a pointer).
The `stackTop` member is initialized with it in C++: https://dxr.mozilla.org/mozilla-central/source/js/public/RootingAPI.h#839.